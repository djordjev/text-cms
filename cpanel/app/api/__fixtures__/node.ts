import { FsNode } from '@prisma/client';

const newNode = (prototype?: Partial<FsNode>, children?: Partial<FsNode>[]) => {
  // eslint-disable-next-line
  const result: any = {
    createdAt: prototype?.createdAt ?? new Date(2024, 2, 6, 12, 12, 12, 12),
    id: prototype?.id ?? 1,
    name: prototype?.name ?? 'random_name',
    parentId: prototype?.parentId ?? 1001,
    path: prototype?.path ?? '/randompath',
    typeId: prototype?.typeId ?? 1
  };

  if (children) {
    result.children = [];

    for (const child of children) {
      const newChild = {
        createdAt: child?.createdAt ?? new Date(2024, 2, 6, 12, 12, 12, 12),
        id: child?.id ?? 1,
        name: child?.name ?? 'random_name',
        parentId: child?.parentId ?? 1001,
        path: child?.path ?? '/randompath',
        typeId: child?.typeId ?? 1
      };

      result.children.push(newChild);
    }
  }

  return result;
};

export { newNode };
