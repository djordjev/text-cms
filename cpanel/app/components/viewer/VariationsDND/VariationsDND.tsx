import {
  closestCenter,
  DndContext,
  KeyboardSensor,
  PointerSensor,
  useSensor,
  useSensors
} from '@dnd-kit/core';
import type { DragEndEvent } from '@dnd-kit/core/dist/types';
import {
  SortableContext,
  sortableKeyboardCoordinates,
  verticalListSortingStrategy
} from '@dnd-kit/sortable';
import { FC } from 'react';

import { DraggableVariation } from '~/components/viewer/DraggableVariation';
import { FileVariation } from '~/types';

export interface VariationsDNDProps {
  id: number;
  onDragEnd: (data: { from: string; to: string }) => void;
  variations: FileVariation[];
}

const VariationsDND: FC<VariationsDNDProps> = (props) => {
  const { id, onDragEnd, variations } = props;

  // Hooks
  const sensors = useSensors(
    useSensor(PointerSensor),
    useSensor(KeyboardSensor, {
      coordinateGetter: sortableKeyboardCoordinates
    })
  );

  // Handlers
  const handleDragEnd = (event: DragEndEvent) => {
    const { active, over } = event;

    if (!over) return;

    onDragEnd({ from: `${active.id}`, to: `${over.id}` });
  };

  return (
    <DndContext
      sensors={sensors}
      collisionDetection={closestCenter}
      onDragEnd={handleDragEnd}
    >
      <SortableContext
        items={variations}
        strategy={verticalListSortingStrategy}
      >
        {variations.map((v) => (
          <DraggableVariation key={v.id} fileId={id} variation={v} />
        ))}
      </SortableContext>
    </DndContext>
  );
};

export { VariationsDND };
