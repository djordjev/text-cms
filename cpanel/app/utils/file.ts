const isHome = (name: string) => {
  return name.toLowerCase() === 'finder';
};

const isFile = (name: string) => {
  return name.endsWith('.txt');
};

const isFolder = (name: string) => {
  return !isFile(name);
};

const getPathFromSegments = (segments: string[], segment: string) => {
  const index = segments.indexOf(segment);

  if (index < 1) return '/finder';

  const mapped = segments.slice(0, index + 1).map((segment) => {
    if (isHome(segment)) return 'finder';

    return segment;
  });

  const joined = mapped.join('/');

  return `/${joined}`;
};

export { getPathFromSegments, isHome, isFile, isFolder };
