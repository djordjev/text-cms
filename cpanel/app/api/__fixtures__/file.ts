import { TEST_EDITOR } from '~/utils/test';

export const FILE_CONTENT = [
  {
    id: '1',
    name: 'First Variation',
    condition: [[['a', '=', '1']]],
    text: TEST_EDITOR
  },
  {
    id: '2',
    name: 'Second Variation',
    condition: [
      [
        ['b', '=', '1'],
        ['c', '>', '12']
      ],
      [['w', '=', 'true']]
    ],
    text: TEST_EDITOR
  }
];

export const FILE_VARIATION = JSON.stringify(FILE_CONTENT);
