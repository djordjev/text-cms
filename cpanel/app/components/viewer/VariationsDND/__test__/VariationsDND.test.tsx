import { createRemixStub } from '@remix-run/testing';
import { render, screen } from '@testing-library/react';

import { FILE_CONTENT } from '~/api/__fixtures__/file';

import { VariationsDND, VariationsDNDProps } from '../VariationsDND';

test('VariationsDND', () => {
  const props: VariationsDNDProps = {
    id: 1,
    onDragEnd: vi.fn(),
    variations: FILE_CONTENT as any
  };

  const createComponent = () => {
    const RemixStub = createRemixStub([
      {
        path: '/',
        Component: () => <VariationsDND {...props} />
      }
    ]);

    return <RemixStub />;
  };

  const { container } = render(createComponent());

  expect(screen.getByText('First Variation')).toBeInTheDocument();

  expect(container.childNodes).toMatchSnapshot();
});
