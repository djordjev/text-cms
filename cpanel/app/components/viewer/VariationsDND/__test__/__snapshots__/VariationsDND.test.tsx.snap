// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`VariationsDND 1`] = `
NodeList [
  <div
    aria-describedby="DndDescribedBy-0"
    aria-disabled="false"
    aria-roledescription="sortable"
    class="u-cursor-move"
    role="button"
    tabindex="0"
  >
    <div
      class="u-card u-shadow-xl u-mb-5x u-bg-base-300 u-p-3xs"
    >
      <div
        class="u-mb-1x u-flex"
      >
        <div
          class="u-flex-1"
        >
          <span
            class="u-italic"
          >
            Variation Name: 
          </span>
          <h4
            class="u-text-secondary-content u-font-bold u-inline"
          >
            First Variation
          </h4>
        </div>
        <svg
          aria-label="move"
          class="tabler-icon tabler-icon-arrows-move"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M18 9l3 3l-3 3"
          />
          <path
            d="M15 12h6"
          />
          <path
            d="M6 9l-3 3l3 3"
          />
          <path
            d="M3 12h6"
          />
          <path
            d="M9 18l3 3l3 -3"
          />
          <path
            d="M12 15v6"
          />
          <path
            d="M15 6l-3 -3l-3 3"
          />
          <path
            d="M12 3v6"
          />
        </svg>
      </div>
      <div
        class="u-flex u-mb-1x"
      >
        <span
          class="u-mr-3x u-italic"
        >
          Conditions:
        </span>
        <div>
          <div>
            <span>
              a
               
              =
               
              1
            </span>
            <span
              class="u-font-bold u-text-secondary-content last:u-hidden"
            >
               
              AND
               
            </span>
          </div>
          <div
            class="u-divider last:u-hidden u-my-1xs"
          >
            OR
          </div>
        </div>
      </div>
      <div>
        <h5
          class="u-italic u-mb-2xs"
        >
          Text:
        </h5>
        <div>
          <h3
            class="u-text-center u-text-lg"
          >
            <span
              class=""
            >
              HEADING
            </span>
          </h3>
          <p
            class=""
          >
            <span
              class=""
            >
              A 
            </span>
            <span
              class="u-italic"
            >
              line
            </span>
            <span
              class=""
            >
               of 
            </span>
            <span
              class="u-font-bold"
              style="color: rgb(228, 17, 17);"
            >
              text
            </span>
            <span
              class=""
            >
               in a 
            </span>
            <span
              class="u-font-bold"
            >
              paragraph
            </span>
            <span
              class=""
            >
              .
            </span>
          </p>
        </div>
      </div>
    </div>
  </div>,
  <div
    aria-describedby="DndDescribedBy-0"
    aria-disabled="false"
    aria-roledescription="sortable"
    class="u-cursor-move"
    role="button"
    tabindex="0"
  >
    <div
      class="u-card u-shadow-xl u-mb-5x u-bg-base-300 u-p-3xs"
    >
      <div
        class="u-mb-1x u-flex"
      >
        <div
          class="u-flex-1"
        >
          <span
            class="u-italic"
          >
            Variation Name: 
          </span>
          <h4
            class="u-text-secondary-content u-font-bold u-inline"
          >
            Second Variation
          </h4>
        </div>
        <svg
          aria-label="move"
          class="tabler-icon tabler-icon-arrows-move"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M18 9l3 3l-3 3"
          />
          <path
            d="M15 12h6"
          />
          <path
            d="M6 9l-3 3l3 3"
          />
          <path
            d="M3 12h6"
          />
          <path
            d="M9 18l3 3l3 -3"
          />
          <path
            d="M12 15v6"
          />
          <path
            d="M15 6l-3 -3l-3 3"
          />
          <path
            d="M12 3v6"
          />
        </svg>
      </div>
      <div
        class="u-flex u-mb-1x"
      >
        <span
          class="u-mr-3x u-italic"
        >
          Conditions:
        </span>
        <div>
          <div>
            <span>
              b
               
              =
               
              1
            </span>
            <span
              class="u-font-bold u-text-secondary-content last:u-hidden"
            >
               
              AND
               
            </span>
            <span>
              c
               
              &gt;
               
              12
            </span>
            <span
              class="u-font-bold u-text-secondary-content last:u-hidden"
            >
               
              AND
               
            </span>
          </div>
          <div
            class="u-divider last:u-hidden u-my-1xs"
          >
            OR
          </div>
          <div>
            <span>
              w
               
              =
               
              true
            </span>
            <span
              class="u-font-bold u-text-secondary-content last:u-hidden"
            >
               
              AND
               
            </span>
          </div>
          <div
            class="u-divider last:u-hidden u-my-1xs"
          >
            OR
          </div>
        </div>
      </div>
      <div>
        <h5
          class="u-italic u-mb-2xs"
        >
          Text:
        </h5>
        <div>
          <h3
            class="u-text-center u-text-lg"
          >
            <span
              class=""
            >
              HEADING
            </span>
          </h3>
          <p
            class=""
          >
            <span
              class=""
            >
              A 
            </span>
            <span
              class="u-italic"
            >
              line
            </span>
            <span
              class=""
            >
               of 
            </span>
            <span
              class="u-font-bold"
              style="color: rgb(228, 17, 17);"
            >
              text
            </span>
            <span
              class=""
            >
               in a 
            </span>
            <span
              class="u-font-bold"
            >
              paragraph
            </span>
            <span
              class=""
            >
              .
            </span>
          </p>
        </div>
      </div>
    </div>
  </div>,
  <div
    id="DndDescribedBy-0"
    style="display: none;"
  >
    
    To pick up a draggable item, press the space bar.
    While dragging, use the arrow keys to move the item.
    Press space again to drop the item in its new position, or press escape to cancel.
  
  </div>,
  <div
    aria-atomic="true"
    aria-live="assertive"
    id="DndLiveRegion-0"
    role="status"
    style="position: fixed; width: 1px; height: 1px; margin: -1px; border: 0px; padding: 0px; overflow: hidden; clip-path: inset(100%); white-space: nowrap;"
  />,
]
`;
